ssrando/ssrando
# Tadtonesanity

Aims:
* Add support for custom items
* Add 17 "Group of Tadtones" items to the pool
* Add 17 checks to Flooded Faron Woods (one for each vanilla tadtone group)
* Add option to toggle Tadtonesanity
* Allow entry into Flooded Great Tree without Water Scale
  * Add a bomb flower on the lilypad in-front of Bucha to prevent softlocks

Details:
* `keysanity.asm` has been renamed to `custom_items.asm`
* Custom functions have been added so that custom items can be more easily added in the future
  * This includes setting custom models for items and should be able to be used for adding the Sailcloth item (with its correct model)
* Logic for entering Flooded Great Tree has been updated to no longer require Water Scale
* As you need to watch the tadtones jingle in Flooded Faron Woods before getting the Water Dragon's Reward check, logic takes into account that you need access to Flooded Faron Woods in addition to Flooded Great Tree
* The Tadtone dowsing cutscene trigger has been removed since tadtone dowsing is a start flag. This can, ofc, be reversed if we ever want to make getting dowsing abilities checks
* The Cranioc cutscene trigger has been removed as it is buggy when getting it after already having gotten the Water Dragon's Reward check

Tests Needed:
* Verify getting a "Group of Tadtones" item from chests, npcs, freestanding, etc works as expected
* Verify the item get textbox shows the correct number of tadtones remaining
* Verify Fi shows you to have the correct number of tadtones
* Verify you can enter and leave Flooded Great Tree via the Water Dragon - even if you don't have Water Scale
* Verify you can get the Water Dragon's Reward check only after getting all 17 tadtones and that you can only get it once
  * You must watch the tadtones jingle in Flooded Faron Woods first (the one cutscene I won't remove xD)
* Verify you can get each tadtone check in Flooded Faron Woods both before getting the Water Dragon's Reward check and after
  * Getting all 17 should **not** start the tadtones jingle
  * The on-screen music scroll should update (even after getting the Water Dragon's Reward check)
  * The Cranoic and Tadtone dowsing cutscenes shouldn't play

Known Issues:
* The preview for the "Group of Tadtones" item in Beedle's Shop is raised higher than usual. This is true of other existing items and cannot be changed right now (not until someone looks into it at least)
